<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from iRobot.urdf.xacro              | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="iRobot" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <!-- MODELS -->
  <material name="iRobot/Green">
    <color rgba="0.0 1.0 0.0 1.0"/>
  </material>
  <material name="iRobot/Yellow">
    <color rgba="1.0 1.0 0.0 1.0"/>
  </material>
  <material name="iRobot/LightYellow">
    <color rgba="1.0 1.0 0.6 1.0"/>
  </material>
  <material name="iRobot/Blue">
    <color rgba="0.0 0.0 1.0 1.0"/>
  </material>
  <material name="iRobot/LightBlue">
    <color rgba="0.6 0.79 1.0 1.0"/>
  </material>
  <material name="iRobot/DarkGrey">
    <color rgba="0.31 0.31 0.31 1.0"/>
  </material>
  <material name="iRobot/LightGrey">
    <color rgba="0.83 0.83 0.83 1.0"/>
  </material>
  <material name="iRobot/MediumGrey">
    <color rgba="0.5 0.5 0.5 1.0"/>
  </material>
  <material name="iRobot/Black">
    <color rgba="0.0 0.0 0.0 1.0"/>
  </material>
  <material name="iRobot/Red">
    <color rgba="1.0 0.0 0.0 1.0"/>
  </material>
  <material name="iRobot/Orange">
    <color rgba="1.0 0.65 0.0 1.0"/>
  </material>
  <link name="base_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
  </link>
  <joint name="base_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="base_link"/>
    <child link="chassis"/>
  </joint>
  <link name="chassis">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="10"/>
      <inertia ixx="0.425" ixy="0.0" ixz="0.0" iyy="0.425" iyz="0.0" izz="0.85"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.08" radius="0.17"/>
      </geometry>
      <material name="iRobot/LightGrey"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.08" radius="0.17"/>
      </geometry>
    </collision>
  </link>
  <joint name="left_axis_base_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0.14 -0.02"/>
    <parent link="chassis"/>
    <child link="left_axis"/>
  </joint>
  <link name="left_axis">
    <inertial>
      <origin rpy="-1.57079632679 0 0" xyz="0 0 0"/>
      <mass value="0.001"/>
      <inertia ixx="6.25e-07" ixy="0.0" ixz="0.0" iyy="6.25e-07" iyz="0.0" izz="2.08333333333e-07"/>
    </inertial>
    <visual>
      <origin rpy="-1.57079632679 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.02" radius="0.0025"/>
      </geometry>
      <material name="iRobot/MediumGray"/>
    </visual>
    <collision>
      <origin rpy="-1.57079632679 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.02" radius="0.0025"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="left_axis_axis">
    <gravity>true</gravity>
    <material>Gazebo/Red</material>
    <mu1>0.8</mu1>
    <mu2>0.8</mu2>
  </gazebo>
  <joint name="left_wheel_rot_joint" type="continuous">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="100" velocity="100"/>
    <joint_properties damping="0.0" friction="0.0"/>
    <parent link="left_axis"/>
    <child link="left_wheel_link"/>
  </joint>
  <link name="left_wheel_link">
    <inertial>
      <origin rpy="-1.57079632679 0 0" xyz="0 0.01 0"/>
      <mass value="0.5"/>
      <inertia ixx="0.00375" ixy="0.0" ixz="0.0" iyy="0.00375" iyz="0.0" izz="0.00125"/>
    </inertial>
    <visual>
      <origin rpy="-1.57079632679 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.02" radius="0.03"/>
      </geometry>
      <material name="iRobot/Black"/>
    </visual>
    <collision>
      <origin rpy="-1.57079632679 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.02" radius="0.03"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="left_wheel_link">
    <gravity>true</gravity>
    <material>Gazebo/Grey</material>
    <mu1>50</mu1>
    <mu2>10</mu2>
    <kp>10000000</kp>
    <kd>1</kd>
    <fdir1>1 0 0</fdir1>
    <minDepth>0.0001</minDepth>
    <maxContacts>1</maxContacts>
  </gazebo>
  <joint name="right_axis_base_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 -0.14 -0.02"/>
    <parent link="chassis"/>
    <child link="right_axis"/>
  </joint>
  <link name="right_axis">
    <inertial>
      <origin rpy="-1.57079632679 0 0" xyz="0 0 0"/>
      <mass value="0.001"/>
      <inertia ixx="6.25e-07" ixy="0.0" ixz="0.0" iyy="6.25e-07" iyz="0.0" izz="2.08333333333e-07"/>
    </inertial>
    <visual>
      <origin rpy="-1.57079632679 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.02" radius="0.0025"/>
      </geometry>
      <material name="iRobot/MediumGray"/>
    </visual>
    <collision>
      <origin rpy="-1.57079632679 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.02" radius="0.0025"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="right_axis_axis">
    <gravity>true</gravity>
    <material>Gazebo/Red</material>
    <mu1>0.8</mu1>
    <mu2>0.8</mu2>
  </gazebo>
  <joint name="right_wheel_rot_joint" type="continuous">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="100" velocity="100"/>
    <joint_properties damping="0.0" friction="0.0"/>
    <parent link="right_axis"/>
    <child link="right_wheel_link"/>
  </joint>
  <link name="right_wheel_link">
    <inertial>
      <origin rpy="-1.57079632679 0 0" xyz="0 0.01 0"/>
      <mass value="0.5"/>
      <inertia ixx="0.00375" ixy="0.0" ixz="0.0" iyy="0.00375" iyz="0.0" izz="0.00125"/>
    </inertial>
    <visual>
      <origin rpy="-1.57079632679 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.02" radius="0.03"/>
      </geometry>
      <material name="iRobot/Black"/>
    </visual>
    <collision>
      <origin rpy="-1.57079632679 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.02" radius="0.03"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="right_wheel_link">
    <gravity>true</gravity>
    <material>Gazebo/Grey</material>
    <mu1>50</mu1>
    <mu2>10</mu2>
    <kp>10000000</kp>
    <kd>1</kd>
    <fdir1>1 0 0</fdir1>
    <minDepth>0.0001</minDepth>
    <maxContacts>1</maxContacts>
  </gazebo>
  <joint name="sphere_joint1" type="revolute">
    <origin rpy="0 0 0" xyz="0.14 0 -0.04"/>
    <axis xyz="0 0 1"/>
    <limit effort="100" lower="0" upper="0" velocity="100"/>
    <parent link="chassis"/>
    <child link="sphere1"/>
  </joint>
  <link name="sphere1">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.3"/>
      <inertia ixx="0.0012" ixy="0.0" ixz="0.0" iyy="0.0012" iyz="0.0" izz="0.0012"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.01"/>
      </geometry>
      <material name="iRobot/Black"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.01"/>
      </geometry>
    </collision>
  </link>
  <joint name="sphere_joint2" type="revolute">
    <origin rpy="0 0 0" xyz="-0.14 0 -0.04"/>
    <axis xyz="0 0 1"/>
    <limit effort="100" lower="0" upper="0" velocity="100"/>
    <parent link="chassis"/>
    <child link="sphere2"/>
  </joint>
  <link name="sphere2">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.3"/>
      <inertia ixx="0.0012" ixy="0.0" ixz="0.0" iyy="0.0012" iyz="0.0" izz="0.0012"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.01"/>
      </geometry>
      <material name="iRobot/Black"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.01"/>
      </geometry>
    </collision>
  </link>
</robot>